build.proj

```diff
@@ -65,7 +65,23 @@
 
   <Import Project="dir.targets" />
   <Import Project="dir.traversal.targets" />
-  <Import Project="$(ToolsDir)clean.targets" />
+  <Import Project="$(ToolsDir)clean.targets" />  
+ 
+  <!-- 
+    This is to initialize the ProjectJsonFiles ItemGroup with the template files during the update dependencies target
+    only to prevent ValidateExactRestore from failing on just the regular project.json files.
+  -->
+  <Target Name="IncludeOnlyTemplateFilesForUpdate" BeforeTargets="UpdateDependencies">
+    <ItemGroup>
+      <ProjectJsonFiles Remove="*" />
```

Comment:
Because the ProjectJsonFiles ItemGroup currently is used in both ValidateExactRestore (which requires actual project.json to look up the lockfile) and UpdateDependencies (which requires the template files to update). I thought a condition would be to separate the two cases